Assets/Mirror/Editor/Weaver/CompilationFinishedHook.cs

```diff
@@ -47,6 +47,29 @@ static void HandleError(string msg)
         static void OnInitializeOnLoad()
         {
             CompilationPipeline.assemblyCompilationFinished += OnCompilationFinished;
+
+            // the first time we load the weaver we need to weave existing assemblies
+            // and reload them.
+            // Store an environment variable that will survive assembly reloads
+            // so that we know if we have run the weaver before
+            if (Environment.GetEnvironmentVariable("MIRROR_WEAVER") != "true" )
+            {
+                Environment.SetEnvironmentVariable("MIRROR_WEAVER", "true");
+                WeaveExisingAssemblies();
+            }
+        }
+
+        static void WeaveExisingAssemblies()
```

Comment:
what are the other options to do this?
e.g. EditorPrefs
EditorSettings
seeing if a certain file in Library folder exists
etc.?

is this the only way?

or is there a 100% reliable way to detect if we weaved before?

Reply:
EditorPrefs is not good,  they are persistent,  so if you close the project, wipe out Library and reopen,  it will think it already ran.
Creating a file also has similar problems.

I 100% realize it is hacky to use environment variable for this,  but I chose it that way because the environment variable will be completely gone once you close unity,  I need something that will survive reloading the dlls  but will not survive restarting unity.

Reply:
what do you mean with 'will be completely gone once you close unity'?
isn't it stored in the library folder somewhere?

Reply:
No,  environment variables are never stored,  they are only kept as part of the process.  Once the process is gone,  so are the environment variables.   That is what I need,  but it looks like 
https://docs.unity3d.com/ScriptReference/SessionState.html  is a better solution that accomplishes the same thing.

Reply:
Fixed using SessionState